### ISSUE TITLE
Define a process for maintaining the py3 branch

### ISSUE THREAD

**Issue by taldcroft** [sot/skare#109]
The py3 (and Ska3) branch is getting behind.  Not clear if there is any easy solution, but we don't want to let this keep sliding too much.

**Comment by jeanconn** [comment]
I wasn't with you in the start of this process, but what requires it to live in a separate branch?

**Comment by taldcroft** [comment]
Mostly `pkgs.conda`, but there are other diffs.  In fact py3 is not even merge-able to master now:  https://github.com/sot/skare/compare/py3.  (There is something wonky about this diff since it implies a lot of non-3.x versions of packages in master in pkgs.manifest).

**Comment by jeanconn** [comment]
Gotcha.  I was wondering about breaking pkgs.conda then into shared list and a non-shared list.

**Comment by jeanconn** [comment]
For the record, @taldcroft somewhat vetoed my "reunification" thought in #116, so now to just define process to keep py3 close to master in package versions.  That may be a step to just check if a commit could also go into py3 and then cherry-pick from master.

**Comment by jeanconn** [comment]
So, are thoughts now that the process will be to rebase the py3-rebase (or whatever) branch on master on all changes in master?

**Comment by taldcroft** [comment]
No, cherry-pick relevant commits to master onto the rebased `py3`.

**Comment by jeanconn** [comment]
OK, I'll start with that as the plan and come back to this if we see other oddities (merge conflicts).

**Comment by taldcroft** [comment]
Answer:

Cherry-pick if possible, rebase from master otherwise (and fix conflict on the way).

**Comment by taldcroft** [comment]
Update, you need the `-m 1` option, e.g.:
```
git cherry-pick -m 1 3c12ac9
```
If you look at the commits in the `py3` branch you will see how this looks.

**Comment by taldcroft** [comment]
Although for some reason then github failed to generate the correct diffs and, e.g., it was showing that the `py3` branch was mv'ing all the NOTES files.  Command line `git diff` was giving the right result.  Hmm.  Maybe rebase is the way...

**Comment by taldcroft** [comment]
Well I just used rebase.  For now that may be the simplest and get us through to the point when py3 is prime and py2 is legacy.

**Comment by jeanconn** [comment]
Are the changes you rebased just in either your repo or the aca repo?  I
don't see an up-to-date py3 branch on github skare and am confused.

**Comment by jeanconn** [comment]
Oh wait, I was just confused because the recent commits ended up "earlier" than the last commit that basically stores your clean-up items between the branches.  Basically, github is telling me the branch was updated 4 months ago when really it has commits from a week ago.  Gotcha.

**Comment by jeanconn** [comment]
Does HEAD ska3/test need to be kept up-to-date too?

**Comment by taldcroft** [comment]
OK now I did regular cherry-pick for the agasc and chandra_aca updates.

**Comment by taldcroft** [comment]
> Does HEAD ska3/test need to be kept up-to-date too?

Probably not.

**Comment by taldcroft** [comment]
I'm creating /proj/sot/ska3/flight on GRETA.  We can start keeping that synced.
